lib/storage/src/content_manager/toc/point_ops.rs

```diff
@@ -246,29 +247,39 @@ impl TableOfContent {
             .map_err(|err| err.into())
     }
 
+    /// # Cancel safety
+    ///
+    /// This method is cancel safe.
     async fn _update_shard_keys(
         collection: &Collection,
         shard_keys: Vec<ShardKey>,
         operation: CollectionUpdateOperations,
         wait: bool,
         ordering: WriteOrdering,
     ) -> Result<UpdateResult, StorageError> {
-        if shard_keys.is_empty() {
-            return Err(StorageError::bad_input("Empty shard keys selection"));
-        }
+        // `Collection::update_from_client` is cancel safe, so this method is cancel safe.
 
-        let updates: Vec<_> = shard_keys
+        // TODO: Is this cancel safe!? I.e., is it safe to cancel *multiple* `update_from_client` requests?
```

Comment:
Do we still need to resolve this?

I'd assume it is **not** cancel safe, because it may apply an operation partially.

Reply:
It is cancel safe, I've described this in the function documentation in a new commit.